pkg/clients/clients.go

```diff
@@ -10,7 +10,7 @@ type InternalDockerClient interface {
 	ExtractNetworksForContainer(containerName string) (map[string]*network.EndpointSettings, error)
 	ConnectContainerToNetworks(containerName string, settings map[string]*network.EndpointSettings) error
 	ConnectContainerToNetworksByNames(containerName string, networkName []string) error
-	StopAndRemoveDockerContainer() error
+	StopDockerContainer(removeContainer bool) error
```

Comment:
try to keep the same function and and use some param like skipRemove. It will be good if this skipRemove is give by hooks only and service shouldn't give this param to hooks and hooks to internalDockerClient.

Reply:
Refactored